service/worker/workerdeployment/client.go

```diff
@@ -525,6 +528,9 @@ func (d *ClientImpl) SetCurrentVersion(
 		},
 	)
 	if err != nil {
+		if isResourceExhausted(err) {
+			return nil, serviceerror.NewInternal(fmt.Sprintf(ErrTooManyRequests, "SetCurrentVersion"))
```

Comment:
If these messages are going to the user we should put the full API name in them otherwise users will be confused.

Also, I wonder if this is for user benefit and not internal error handling, it might make most sense to put the conversion in frontend handlers.